drivers/ieee802154/ieee802154_nrf5.c

```diff
@@ -1163,19 +1173,27 @@ void nrf_802154_energy_detected(const nrf_802154_energy_detected_t *result)
 {
 	if (nrf5_data.energy_scan_done != NULL) {
 		energy_scan_done_cb_t callback = nrf5_data.energy_scan_done;
-
+#if defined(CONFIG_IEEE802154_RAW_MODE)
+		const struct device *dev = nrf5_dev;
+#else
+		const struct device *dev = net_if_get_device(nrf5_data.iface);
+#endif
 		nrf5_data.energy_scan_done = NULL;
-		callback(net_if_get_device(nrf5_data.iface), result->ed_dbm);
+		callback(dev, result->ed_dbm);
 	}
 }
 
 void nrf_802154_energy_detection_failed(nrf_802154_ed_error_t error)
 {
 	if (nrf5_data.energy_scan_done != NULL) {
 		energy_scan_done_cb_t callback = nrf5_data.energy_scan_done;
-
+#if defined(CONFIG_IEEE802154_RAW_MODE)
+		const struct device *dev = nrf5_dev;
+#else
+		const struct device *dev = net_if_get_device(nrf5_data.iface);
+#endif
```

Comment:
Could this pattern be wrapped in a function like
```c
static const struct device * nrf5_get_device(void)
{
#if defined(CONFIG_IEEE802154_RAW_MODE)
    return nrf5_dev;
#else
    return net_if_get_device(nrf5_data.iface);
#endif
}
```
and when we need this w could use unconditionally:
```c
   const struct device *dev = nrf5_get_device();
```

Reply:
Thank you for the suggestion, it's applied now.